### ISSUE TITLE
Deserialization ignores '>' characters

### ISSUE THREAD

**Issue by Galapen** [WasabiThumb/minimessage-js#5]
Firstly want to say thanks for the great lib!

I'm a bit confused on this one. Using npm release 1.1.2, attempting the following:
```ts
const mm = MiniMessage.miniMessage()
mm.deserialize("<gold>Test>>>>>").toJSON()
```
Outputs:
```json
{
    "text": "Test",
    "color": "gold"
}
```
Showing only "Test" in gold when using `toHTML`; omitting the `>` characters entirely.
![image](https://github.com/user-attachments/assets/fcc8d6ea-43a7-4ef0-8b71-a4328edf7e82)

But when entered into your demo, the JSON shows:
```json
{
 "text": "Test>",
 "extra": [
  ">",
  ">",
  ">",
  ">"
 ],
 "color": "gold"
}
```
![image](https://github.com/user-attachments/assets/c127bc60-e16f-494b-903b-cf815b53f0c5)

The demo displays the content correctly. I could well be doing something wrong, but I'm hoping you can verify.

**Comment by WasabiThumb** [comment]
That's very strange. [This line](https://github.com/WasabiThumb/minimessage-js/blob/ea493329339589fdaa807bddf9b34f9ce0de8c04/src/markup/parser.ts#L108) specifically exists to prevent this issue.
> <sub>When a ``>`` character is read and a tag is NOT being read (no corresponding ``<`` or any ``<`` has already been closed) it is appended to the plain text buffer, and then [emitted](https://github.com/WasabiThumb/minimessage-js/blob/ea493329339589fdaa807bddf9b34f9ce0de8c04/src/markup/parser.ts#L168) in the ``text`` event. This also explains why all of the ``>`` characters appear in ``extra`` separately, as it causes the buffer to be flushed each time. A harmless quirk.</sub>

It was added on August 2nd, the same day as the 1.1.2 NPM release. Though when simply [reading the built code via NPM](https://www.npmjs.com/package/minimessage-js/v/1.1.2?activeTab=code), it's obvious that the check isn't there:
![image](https://github.com/user-attachments/assets/a1b3e930-a9cb-4b8f-ba74-55cbf96a7eae)

TL;DR, I blundered. I'll publish a new release shortly.

**Comment by WasabiThumb** [comment]
[Release 1.1.3 has been published](https://www.npmjs.com/package/minimessage-js), let me know if you still have issues.
